expr/literals.go

```diff
@@ -406,21 +463,68 @@ type ProtoLiteral struct {
 	Type  types.Type
 }
 
+func timeFromPrecisionUnits(units int64, precision types.TimePrecision) time.Time {
```

Comment:
I wonder if we could hang a `ToTime(value)` method on `types.TimePrecision` or we could hang `ToTime(precision)` on `types.TimestampTz` and friends....

Reply:
done